### PR TITLE
Feature/numeral

### PR DESCRIPTION
**Pull request by tnagorra**

### CODE REVIEW

File: src/components/Numeral/index.tsx [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+    const output = formatNumber(
+        value,
+        separator,
+        abbreviate,
+        precise,
+    );
+
+    return (
+        <span className={className}>
+            {`${prefix}${output}${suffix}`}
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx]: We should allow this to be styled as well

File: src/components/Numeral/index.tsx [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+    // TODO: do not hardcode decimal symbol
+
+    const indexOfDecimal = output.indexOf('.');
+    if (indexOfDecimal !== -1) {
+        if (/\.0+$/.test(output)) {
+            console.warn('there', output);
+            // Remove all trailing zeros starting immediately after decimal
+            output = output.substr(0, indexOfDecimal);
+        } else {
+            console.warn('here', output);
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx]: We should not leave `console.warn` here and there

File: src/components/Numeral/index.tsx [hunk 3]
```diff
@@ -0,0 +1,146 @@
...
+    const precise = precision < 0 ? getAutoPrecision(value) : precision;
+    const output = formatNumber(
+        value,
+        separator,
+        abbreviate,
+        precise,
+    );
+
+    return (
+        <span className={className}>
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx]: return type should be consistent. i.e. should not be `div` in one case and `span` in another

File: src/components/Numeral/index.tsx [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+        return 0;
+    }
+
+    const absoluteValue = Math.abs(value);
+    if (absoluteValue < 1) {
+        return Math.ceil(-Math.log10(absoluteValue)) + 1;
+    }
+
+    // NOTE: ignore precision for large numbers
+    if (absoluteValue > 100) {
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx]: User should be able to customize this

File: src/components/Numeral/index.tsx [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+            // Remove trailing zeros after decimal
+            output = output.replace(/(\.\d*[1-9]+)0+/, '$1');
+        }
+    }
+
+    if (isTruthyString(separator)) {
+        output = addSeparator(output, separator);
+    }
+
+    return `${output}${suffix ?? ''}`;
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx]: User should be able to get the output and suffix separately

File: src/components/Numeral/index.tsx [hunk 6]
```diff
@@ -0,0 +1,146 @@
...
+    */
+    placeholder?: string;
+}
+
+/**
+ * Number formatting component
+ */
+function Numeral({
+    value,
+    precision = -1,
```
Comment (frozenhelium) [inline src/components/Numeral/index.tsx:146]: should be added to documentation if value of `-1` causes auto precision

Comment (tnagorra) [reply src/components/Numeral/index.tsx:146]: It is well documented.

**Review by frozenhelium — APPROVED** [review]

**Pull request merged**